### PR TITLE
Add test files and regression build system

### PR DESCRIPTION
**Pull request by TiejunMS**

### CODE REVIEW

File: externals/threadx
Comment (bo-ms) [inline externals/threadx:1]: can we avoid including externals folder?

File: scripts/sdl_check.sh
Comment (bo-ms) [inline scripts/sdl_check.sh:1]: do we still need sdl_check.sh and update_patch.sh?

Comment (TiejunMS) [reply scripts/sdl_check.sh:1]: No hurt to put it here. We can remove them if no longer needed in the future.

File: .github/workflows/regression_test.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    - name: Generate Code Coverage Results Summary
+      uses: irongut/CodeCoverageSummary@v1.3.0
+      with:
+        filename: ./test/cmake/coverage_report/default_build_coverage.xml
+        format: markdown
+        badge: true
+        hide_complexity: true
+        output: file
+
```
Comment (ting-ms) [inline .github/workflows/regression_test.yml:75]: Recommend to add following actions to add code coverage summary and checkrun:
```yml
    - name: Write Code Coverage Summary
      run: |
        echo "## Coverage Report" >> $GITHUB_STEP_SUMMARY
        cat code-coverage-results.md >> $GITHUB_STEP_SUMMARY
    - name: Create CheckRun for Code Coverage
      uses: LouisBrunner/checks-action@v1.6.2
      with:
        token: ${{ secrets.GITHUB_TOKEN }}
        name: Code Coverage
        conclusion: ${{ job.status }}
        output: |
          {"summary":"Coverage Report"}
        output_text_description_file: code-coverage-results.md
```

Comment (TiejunMS) [reply .github/workflows/regression_test.yml:75]: Added

**Review by xiuwencai — APPROVED** [review]

**Pull request merged**